### DISCUSSION TITLE
alpha extract fails (on 7.1.1-8)

### DISCUSSION THREAD

**Discussion by indexofrefraction** [ImageMagick/ImageMagick#6299] [Help] [answered]
### ImageMagick version

7.1.1-8 (installed by brew)

### Operating system, version and so on

macos 11.7.5 (Big Sur)

### Description

hi, 

i'm trying to **extract the alpha channel** of the attached tif.
this was working in april 22 but now (may 23 on 7.1.1-8) i just get a **white image**
it would be very nice if somebody could check what is going wrong here...

best, index

`convert icon_0078.tif -alpha extract -depth 8 icon_0078_alpha.tif`

[icon_0078.tif.zip](https://github.com/ImageMagick/ImageMagick/files/11411964/icon_0078.tif.zip)

**Comment by indexofrefraction** [comment]
follow up... it seems this tif has the "alpha" channel stored as Meta0 
macos spotlight shows the sample tif with transparency. photoshops does not, but shows the alpha channel

`convert icon_0078.tif -channel Meta0 -depth 8 icon_0078_alpha.tif` --> gives the desired output

so far so good... but i need to **concatenate multiple icons, alpha channels only**

`montage -mode concatenate -tile 4x -alpha extract -depth 8 icon_0*.tif icons.tif`
this worked in 2022-04 but fails now 

`montage -mode concatenate -tile 4x -channel Meta0 -depth 8 icon_0*.tif icons.tif`
sadly -channel Meta0 does not work in this context 

any input on how to solve this would be great !

**Comment by snibgo** [comment]
exiftool says:
```
Bits Per Sample                 : 8 8 8 8
Extra Samples                   : Unspecified
```
So the metadata says that the fourth channel is *not* alpha. Thus, recent versions of IM correctly record this as a meta channel.

Perhaps ...
```
magick icon_0078.tif -channel meta0 -separate x.png
```
... does what you need.

**Reply by indexofrefraction** [reply, marked as answer]
thanks snibgo,

i could finally solve this by altering the input files like this :
`convert -channel-fx 'meta0=>alpha' input.tif input.RGBA.tif`
after that processing with montage concentate worked again like in image magick 6

still thanks for answering, image magick needs quite some experience
and it helps a lot to get some hints and leads here!

**Comment by snibgo** [comment]
You have IM v7, but are using the (very old) v6 syntax. I suggest:
```
magick input.tif -channel-fx 'meta0=>alpha' input.RGBA.tif
```
Note that this reads the input, then processes it, then writes the output. With this syntax, operations are always performed in the order given on the command.

**Reply by indexofrefraction** [reply]
thanks, i changed that in my script.
